[IMP] mail: redesign the chat views

Brieuc-brd opened the pull request:
This PR redesigns the content in the ChatWindow, Discuss and
the Chatter. The purpose is to make the chats more user-friendly.

Requires: 
- https://github.com/odoo/odoo/pull/95417

task-2823627

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/92872).

Brieuc-brd commented:
Hello @alexkuhn @seb-odoo 👋 

This PR is finally ready for review 🙂 
Could you take a look at it?

cc. @stefanorigano 👀

alexkuhn commented on addons/mail/static/src/components/message_in_reply_to_view/message_in_reply_to_view.xml:
```diff
@@ -2,22 +2,27 @@
-                            <span class="fst-italic me-2">Click to see the attachments</span>
-                            <i class="fa fa-image"/>
-                        </t>
+                    <span class="o_MessageInReplyToView_wrapOuter d-flex align-items-center text-muted opacity-75 opacity-100-hover o_cursor_pointer" t-attf-class="{{ messageInReplyToView.messageView.inChatWindow and messageInReplyToView.messageView.message.isCurrentUserOrGuestAuthor ? 'pe-3' : 'ps-3' }}" t-on-click="messageInReplyToView.onClickReply">
+                        <img class="o_MessageInReplyToView_avatar me-2 rounded-circle" t-att-src="messageInReplyToView.messageView.message.parentMessage.avatarUrl" t-att-title="messageInReplyToView.messageView.message.parentMessage.authorName" alt="Avatar"/>
+                        <span class="o_MessageInReplyToView_wrapInner overflow-hidden">
+                            <b class="o_MessageInReplyToView_author">@<t t-out="messageInReplyToView.messageView.message.parentMessage.authorName"/></b>
+                            <span class="o_MessageInReplyToView_body text-break">
+                                <t t-if="messageInReplyToView.hasBodyBackLink">
+                                    <t t-out="messageInReplyToView.messageView.message.parentMessage.prettyBodyAsMarkup"/>
+                                </t>
+                                <t t-if="messageInReplyToView.hasAttachmentBackLink">
+                                    <span class="me-2 fst-italic">Click to see the attachments</span>
+                                    <i class="fa fa-image"/>
+                                </t>
+                            </span>
```
As said in a prior comment: make a dedicated commit just for change of style for reply

alexkuhn commented on addons/mail/static/src/components/message/message.xml:
```diff
@@ -53,20 +52,22 @@
+                            <div class="o_Message_header d-flex flex-wrap align-items-baseline" t-att-class="{ 'justify-content-end': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor }">
                                 <t t-if="messageView.message.author">
-                                    <strong class="o_Message_authorName o_redirect me-2 text-truncate" t-att-class="{'o_cursor_pointer': messageView.hasAuthorOpenChat}" t-att-role="messageView.hasAuthorOpenChat ? 'button' : ''" t-att-tabindex="messageView.hasAuthorOpenChat ? '0' : ''" t-on-click="messageView.onClickAuthorName" t-att-title="messageView.authorTitleText">
+                                    <strong class="o_Message_authorName o_redirect me-1 text-truncate" t-att-class="{ 'o_cursor_pointer': messageView.hasAuthorOpenChat, 'd-none': messageView.inChatWindow and (messageView.message.isCurrentUserOrGuestAuthor or (messageView.messageListViewMessageViewItemOwner and  messageView.messageListViewMessageViewItemOwner.messageListViewOwner.thread.channel_ty ...
```
Replace `d-none` by `t-if`

alexkuhn replied:
Also the condition is long and hard to understand at a glance.
Easily fixed by introducing the proper concept, e.g. with new (computed) field `MessageView/isShowingAuthorName`

alexkuhn commented on addons/mail/static/src/components/message/message.xml:
```diff
@@ -111,54 +112,111 @@
-                                    className="'o_Message_composer mb-1'"
-                                    record="messageView.composerViewInEditing"
+                        <div 
+                            class="o_Message_bubble_wrap position-relative d-flex align-items-start"
```
rename `bubble_wrap` => `bubbleWrap` (camelCase)

alexkuhn commented on addons/mail/static/src/components/message/message.xml:
```diff
@@ -111,54 +112,111 @@
+                                'ps-5': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor and !messageView.composerViewInEditing,
+                                'pe-2': messageView.inChatWindow and !messageView.message.isCurrentUserOrGuestAuthor and messageView.composerViewInEditing,
+                                'ps-2': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor and messageView.composerViewInEditing,
+                                'o_Message_can_be_deleted': messageView.message.canBeDeleted,
```
`o_Message_can_be_deleted` looks unused

alexkuhn commented on addons/mail/static/src/components/message/message.scss:
```diff
@@ -52,21 +50,25 @@
-        border-bottom: 1px solid darken(map-get($grays, '400'), 5%);
+    &.o-reply-highlighted {
+        transform: translateY(-#{map-get($spacers, 3)});
     }
 
-    &.o-highlighted {
-        background-color: rgba($o-brand-primary, .1);
+    .o-not-current-author {
+        opacity: .15;
+    }
```
We did not follow our naming convention for these modifier classnames.
Since you are introducing and changing them, would camelCase e.g. `o-isReplyHighlighted` and `o-isAuthorNotCurrentUserOrGuest`

alexkuhn commented on addons/mail/static/src/components/message/message.xml:
```diff
@@ -3,46 +3,45 @@
-                <MessageInReplyToView t-if="messageView.messageInReplyToView" record="messageView.messageInReplyToView"/>
-                <div t-if="messageView.isActive and messageView.messageActionList" class="o_Message_actionListContainer position-absolute top-0 end-0 mt-n3 px-3" t-att-class="{ 'o-squashed': messageView.isSquashed }">
-                    <MessageActionList record="messageView.messageActionList"/>
+                <div class="o_Message_InReplyTo_wrap" t-att-class="{ 'me-5': messageView.inDiscuss or (messageView.inChatWindow and !messageView.message.isCurrentUserOrGuestAuthor), 'ms-5': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor, 'd-flex justify-content-end': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor }">
```
rename `o_Message_InReplyTo_wrap` to `o_Message_inReplyToWrap`

alexkuhn commented on addons/mail/static/src/components/message/message.xml:
```diff
@@ -111,54 +112,111 @@
+                        <div 
+                            class="o_Message_bubble_wrap position-relative d-flex align-items-start"
+                            t-att-class="{
+                                'd-flex justify-content-end': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor,
+                                'pe-5': messageView.inChatWindow and !messageView.message.isCurrentUserOrGuestAuthor and !messageView.composerViewInEditing,
+                                'ps-5': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor and !messageView.composerViewInEditing,
+                                'pe-2': messageView.inChatWindow and !messageView.message.isCurrentUserOrGuestAuthor and messageView.composerViewInEditing,
+                                'ps-2': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor and messageView.composerViewInEditing,
+                                'o_Message_can_be_deleted': messageView.message.canBeDeleted,
```
The template would be easier to understand conditional styling if you introduce concept of left/right alignment in models, e.g. `MessageView/isAlignedLeft` or `MessageView/isAlignedRight`.

**This comment applies to whole commit**

alexkuhn replied:
Ok for me, field name is a bit long, but I much prefer very explicit name than ambiguous or missing information, so that's good enough!

Brieuc-brd commented on addons/mail/static/src/components/message_in_reply_to_view/message_in_reply_to_view.scss:
```diff
@@ -1,17 +1,27 @@
+    &.o-align-right {
+        right: $o-mail-message-sidebar-width / 2;
+        border-radius: 0 $o-mail-rounded-rectangle-border-radius-sm 0 0;
+    }
```
I already added the necessary models in the `MessageView` (https://github.com/odoo/odoo/pull/92872#discussion_r931137410)

What do you think ?

Brieuc-brd commented:
Hello @alexkuhn 👋 ,
Thanks for your feedback, I updated the branch 👍 

What do you think ?

alexkuhn commented on addons/web/static/src/legacy/scss/bootstrap_overridden.scss:
```diff
@@ -261,3 +261,6 @@ $form-switch-checked-bg-image: url("data:image/svg+xml,<svg xmlns='http://www.w3
 $badge-padding-x: 0.4em !default;
+
+// Placeholder color 
+$input-placeholder-color: $gray-300 !default;
```
This affects all input placeholders, so to see with @stefanorigano whether this is ok

stefanorigano replied:
Thanks @alexkuhn .
@Brieuc-brd , I missed this notification. `$gray-300` is sensible lower than default (600)... have you tested the entire UI?

Brieuc-brd replied:
@stefanorigano Yes, this variable seems to be used only for inputs with the `form-control` class. 
https://github.com/odoo/odoo/blob/d108db36ca2974179ded929370ea45a106f2e08c/addons/web/static/lib/bootstrap/scss/forms/_form-control.scss#L57

The purpose of this change is to have the same placeholder color with or without this class to keep consistency.
eg. input with placeholder without the `form-control` class (Project app):
![Capture d’écran 2022-07-29 à 14 45 01](https://user-images.githubusercontent.com/80679690/181764221-fa50374e-93c9-43f0-b383-d5bcf963245d.png)

alexkuhn commented on addons/mail/static/src/models/attachment_list.js:
```diff
@@ -71,6 +71,66 @@ registerModel({
         _computeViewableAttachments() {
             return replace(this.attachments.filter(attachment => attachment.isViewable));
         },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsDiscuss() {
+            return Boolean(
+                (this.messageViewOwner && this.messageViewOwner.isDiscuss) ||
+                (this.composerViewOwner && this.composerViewOwner.isDiscuss)
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsChatWindow() {
+            return Boolean(
+                (this.messageViewOwner && this.messageViewOwner.isChatWindow) ||
+                (this.composerViewOwner && this.composerViewOwner.isChatWindow)
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsChatter() {
+            return Boolean(
+                (this.messageViewOwner && this.messageViewOwner.isChatter) ||
+                (this.composerViewOwner && this.composerViewOwner.isChatter)
+            );
+        },
```
Missing `in` in-between `is` and `Discuss/ChatWindow/Chatter`
An Attachment List is no a X, but is in a X

alexkuhn commented on addons/mail/static/src/models/composer_view.js:
```diff
@@ -903,6 +903,36 @@ registerModel({
             }
             return clear();
         },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsDiscuss() {
+            return Boolean(
+                (this.threadView && (this.threadView.threadViewer.discuss || this.threadView.threadViewer.discussPublicView)) ||
+                (this.messageViewInEditing && (this.messageViewInEditing.discuss || this.messageViewInEditing.discussPublicView))
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsChatWindow() {
+            return Boolean(
+                (this.threadView && this.threadView.threadViewer.chatWindow) ||
+                (this.messageViewInEditing && this.messageViewInEditing.chatWindow)
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsChatter() {
+            return Boolean(
+                (this.threadView && this.threadView.threadViewer.chatter) ||
+                (this.messageViewInEditing && this.messageViewInEditing.isChatter)
+            );
+        },
```
Same remark as before: missing `in` in-between `is` and `X`

alexkuhn commented on addons/mail/static/src/models/message_view.js:
```diff
@@ -267,6 +267,82 @@ registerModel({
                 )
             );
         },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsDiscuss() {
+            return Boolean(
+                this.messageListViewMessageViewItemOwner &&
+                (
+                    this.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.threadViewer.discuss ||
+                    this.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.threadViewer.discussPublicView
+                )
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsChatWindow() {
+            return Boolean(
+                this.messageListViewMessageViewItemOwner &&
+                this.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.threadViewer.chatWindow
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsChatter() {
+            return Boolean(
+                this.messageListViewMessageViewItemOwner &&
+                this.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.threadViewer.chatter
+            );
+        },
```
Same remark: missing `in` keyword

alexkuhn commented on addons/mail/static/src/components/message/message.scss:
```diff
@@ -58,15 +56,21 @@
-        &.o_Message_active {
-            background-color: rgba($o-brand-primary, .2);
-        }
+    .o_Message_prettyBody a.o_mail_redirect {
+        padding: map-get($spacers, 1);
+        border-radius: $border-radius;
+        background: rgba($primary, .1);
```
unflatten, i.e. move it away from `.o_Message` rule and make it a top-level rule starting with `o_Message_prettyBody` immediately

alexkuhn commented on addons/mail/static/src/models/message_in_reply_to_view.js:
```diff
@@ -44,6 +44,10 @@ registerModel({
         _computeHasBodyBackLink() {
             return !this.messageView.message.parentMessage.isBodyEmpty;
         },
+        /**
+         * @private
+         * @returns {boolean}
+         */
     },
```
useless diff

alexkuhn reviewed (commented):
Comment just from looking at the code

alexkuhn commented:
Small remarks on visual:
1. chat window message should stick to message
<img width="788" alt="chat-window-message-action-list-should-stick-to-message" src="https://user-images.githubusercontent.com/6569390/181286292-479978e6-8139-4f0f-9a26-759513b258ef.png">
2. mention style should pop more visually
<img width="289" alt="mention-not-visible-enough" src="https://user-images.githubusercontent.com/6569390/181286304-22b15c85-1a15-443a-87f0-1392bece9a3e.png">
3. message action list should pop more visually
<img width="693" alt="message-action-list-not-visible-enough" src="https://user-images.githubusercontent.com/6569390/181286308-269a4d72-3f82-44d1-b6c4-83cd7f21d457.png">
4. message action list anchor should have minimal gap with scrollbars
<img width="342" alt="message-action-list-too-close-to-scrollbar" src="https://user-images.githubusercontent.com/6569390/181286313-a8db512b-aeca-436e-86e7-bb478717424c.png">

Brieuc-brd commented:
> Small remarks on visual:
> 
> 1. chat window message should stick to message
> 
> <img alt="chat-window-message-action-list-should-stick-to-message" width="788" src="https://user-images.githubusercontent.com/6569390/181286292-479978e6-8139-4f0f-9a26-759513b258ef.png">
> 
> 2. mention style should pop more visually
> <img alt="mention-not-visible-enough" width="289" src="https://user-images.githubusercontent.com/6569390/181286304-22b15c85-1a15-443a-87f0-1392bece9a3e.png">
> 
> 3. message action list should pop more visually
> <img alt="message-action-list-not-visible-enough" width="693" src="https://user-images.githubusercontent.com/6569390/181286308-269a4d72-3f82-44d1-b6c4-83cd7f21d457.png">
> 
> 4. message action list anchor should have minimal gap with scrollbars
> <img alt="message-action-list-too-close-to-scrollbar" width="342" src="https://user-images.githubusercontent.com/6569390/181286313-a8db512b-aeca-436e-86e7-bb478717424c.png">

Hello @alexkuhn 👋 ,
I updated the PR.

About the visual remarks :

1. As we don't have enough space in the `ChatWindow`, we have to place the `MessageActionList` in front of the message when it’s too big (absolute position). 
If we stick the menu to the message, it will be partially hidden when the message is too long.
![demo](https://user-images.githubusercontent.com/80679690/181463187-d8cc2dcf-f3b1-483d-ad0e-f07f4fa28587.jpg)

To get a correct result, we should probably detect the length of the message to know in which position the menu should be. Or maybe with a css grid ? I did some tests but didn't get the expected result.
Maybe this could be an improvement for later, what do you think?

2. Opacity increased from `.1` to `.2` ✅ 

3. I added a light border, but I don't think we should do more to increase the contrast. This menu should not be more visible than the message bubble.

4.Fixed ✅

alexkuhn commented:
Ok for keeping float message action list in chat window, because managing proper stick and overflow is quite tricky, and I prefer postponing this part until we have more awareness of UI data in JS modelling

alexkuhn commented on addons/mail/static/src/components/message/message.scss:
```diff
@@ -52,21 +50,29 @@
 
-    &.o-highlighted {
-        background-color: rgba($o-brand-primary, .1);
+    .o-isAuthorNotCurrentUserOrGuest {
+        opacity: .15;
+    }
```
should be
```scss
.o_Message_background {
    &.o-isAuthorNotCurrentUserOrGuest {
    }
}
```

alexkuhn commented on addons/mail/static/src/components/message/message.xml:
```diff
@@ -3,46 +3,44 @@
-                <MessageInReplyToView t-if="messageView.messageInReplyToView" record="messageView.messageInReplyToView"/>
-                <div t-if="messageView.isActive and messageView.messageActionList" class="o_Message_actionListContainer position-absolute top-0 end-0 mt-n3 px-3" t-att-class="{ 'o-squashed': messageView.isSquashed }">
-                    <MessageActionList record="messageView.messageActionList"/>
+                <div class="o_Message_inReplyToWrap" t-att-class="{ 'me-5': messageView.isInDiscuss or messageView.isInChatWindowAndIsAlignedLeft, 'ms-5': messageView.isInChatWindowAndIsAlignedRight, 'd-flex justify-content-end': messageView.isInChatWindowAndIsAlignedRight }">
```
can group `ms-5` and `d-flex justify-content-end` together (they have exact same condition)

alexkuhn commented on addons/mail/static/src/components/message/message.xml:
```diff
@@ -3,46 +3,44 @@
-                    'mt-3': !messageView.isSquashed and messageView.messageListViewMessageViewItemOwner,
-                    'bg-white': messageView.message.isDiscussionOrNotification and !messageView.isSelected and !(messageView.message.isHighlighted or messageView.isHighlighted) and !messageView.isActive,
-                    'bg-100': (!messageView.message.isDiscussionOrNotification and !messageView.isActive and !messageView.isSelected) or (messageView.isActive and !messageView.isSelected and messageView.message.isDiscussionOrNotification and !(messageView.message.isHighlighted or messageView.isHighlighted)),
-                    'bg-200': !messageView.message.isDiscussionOrNotification and messageView.isActive and !messageView.isSelected,
-                    'bg-400': messageView.isSelected and !(messageView.message.isHighlighted or messageView.isHighlighted),
+                    'o-currentAuthor': messageView.message.isCurrentUserOrGuestAuthor,
+                    'mt-1': !messageView.isSquashed,
```
Can group `o-squashed pt-1` and `mt-1` together (same condition)

Brieuc-brd replied:
`mt-1` has the opposite condition (`messageView.isSquashed` / `!messageView.isSquashed`).
We should not group them :)

alexkuhn replied:
oops, nevermind 🤡

alexkuhn commented on addons/mail/static/src/components/message/message.xml:
```diff
@@ -111,54 +111,110 @@
+                                        'rounded-start-3': messageView.isInChatWindowAndIsAlignedRight,      
+                                        'rounded-bottom-3 border': messageView.message.isDiscussionOrNotification,
+                                        'rounded-3 opacity-50': !messageView.message.isDiscussionOrNotification,
+                                        'o-isAuthorNotCurrentUserOrGuest border-info bg-info-light': !messageView.message.isCurrentUserOrGuestAuthor and !messageView.message.isHighlighted and messageView.message.isDiscussionOrNotification, 
+                                        'border-success bg-success-light opacity-25': messageView.message.isCurrentUserOrGuestAuthor and !messageView.message.isHighlighted and messageView.message.isDiscussionOrNotification,
```
Can group `o-isAuthorNotCurrentUserOrGuest` and `border-success bg-success-light opacity-25` together (same condition)

Brieuc-brd replied:
Same as the previous one, `border-success ...` has `messageView.message.isCurrentUserOrGuestAuthor` in its condition while `o-isAuthorNotCurrentUserOrGuest` has the opposite.

alexkuhn replied:
Indeed... ignore comment then

I have a hard time to see difference when there's `!` prefix on long expressions... I quite dislike the use of this transitive way of expression opposite of `x.y.z`, why language is not just `x.y.!z`😢

alexkuhn commented:
Some visual remarks in chatter:
1. I expect all messages to have a bubble. Maybe have basic greyed bubble
![all-message-should-have-bubble-including-sms](https://user-images.githubusercontent.com/6569390/181493167-647d94ef-c9df-41c7-9a5c-317222601420.png)
2. message action list should stick to message in chatter
![message-action-list-in-chatter-should-stick-to-message](https://user-images.githubusercontent.com/6569390/181493174-cc1a31f9-2a2f-43f6-b895-841a7d82d408.png)

Brieuc-brd commented:
> Some visual remarks in chatter:
> 
> 1. I expect all messages to have a bubble. Maybe have basic greyed bubble
>    ![all-message-should-have-bubble-including-sms](https://user-images.githubusercontent.com/6569390/181493167-647d94ef-c9df-41c7-9a5c-317222601420.png)
> 2. message action list should stick to message in chatter
>    ![message-action-list-in-chatter-should-stick-to-message](https://user-images.githubusercontent.com/6569390/181493174-cc1a31f9-2a2f-43f6-b895-841a7d82d408.png)

1. To keep the consistency, I kept the same colors for the sms 👍 .
![Capture d’écran 2022-07-28 à 14 22 09](https://user-images.githubusercontent.com/80679690/181503777-85c537cb-0a03-4159-81f5-8db082b30e04.png)

2. Same issue as for the ChatWindow (https://github.com/odoo/odoo/pull/92872#issuecomment-1197853554).
Keeping it sticky would create space and alignment issues (the number of icons is not always the same for all messages and could create an alignment inconsistency ) for large messages.
![demo_2](https://user-images.githubusercontent.com/80679690/181503987-07581d62-9111-491f-b65d-4af42a9a2e86.jpg)

alexkuhn commented:
Log note missing bubble:
![Screenshot 2022-07-28 at 14 35 39](https://user-images.githubusercontent.com/6569390/181506325-affd6bdb-0e3d-448e-811e-91ccb8d5c6f0.png)

I think would make sense to have grey bubble for this one.
SMS is okay to keep same style as conversation, because there's the SMS icon on top of message.
Log note, however, there was just background being greyed, so make sense to have different bubble style?

Brieuc-brd commented:
> Log note missing bubble: ![Screenshot 2022-07-28 at 14 35 39](https://user-images.githubusercontent.com/6569390/181506325-affd6bdb-0e3d-448e-811e-91ccb8d5c6f0.png)
> 
> I think would make sense to have grey bubble for this one. SMS is okay to keep same style as conversation, because there's the SMS icon on top of message. Log note, however, there was just background being greyed, so make sense to have different bubble style?

After internal discussion, this choice was made to avoid adding another bubble color and to keep the rendering less "noisy" as possible.
Moreover, it makes the messages more visible in the thread.

alexkuhn commented:
> After internal discussion, this choice was made to avoid adding another bubble color and to keep the rendering less "noisy" as possible. Moreover, it makes the messages more visible in the thread.

IMO this is not a good choice.
When you have bubble layout, I expect all exchange of messages (notes is an internal exchange of messages) to all have this bubble visual.

So I pretty much disagree with this decision 🫤

Brieuc-brd commented:
@alexkuhn,
I just fixed a small condition issue due to the last push and added a `text-muted` on log notes to increase the contrast with a regular message.

alexkuhn commented:
> If I'm talking with a client I prefer to see my notes and activities designed very differently from the messages I'm exchanging with him.

Agree, it's important to distinguish both, because the context is quite different.
TBH I think it's hard to make it different while having a consistent shape e.g. with also bubble layout.
So maybe it's the best trade-off to not have bubble visual for notes.

It's a shame nonetheless, because most internal communication we do is with notes, so we won't have bubble for the most part. Also notes are quite noisy: never squashed, so they always show avatar + name for each of them.

I'm wondering whether the exchange of notes could be easier to see with a change of conversation background just for them... And also prevent the squash.

Here is an idea:
1. currently in this branch:
![chat-window-current](https://user-images.githubusercontent.com/6569390/181586988-383b6b97-af14-45d8-85c3-406e5bc0d226.png)
2. suggestion:
![chat-window-suggestion](https://user-images.githubusercontent.com/6569390/181586993-ee7d0d2f-bd84-4880-a9e7-4caa0da5a113.png)



Personally, I really like when I can understand the overall shape just at a glance. That's why having some visual clues like this would really improvement usability for me. Thanks!

Brieuc-brd commented:
Hello @alexkuhn 👋 ,

I have updated the rendering of the log notes:
- Same padding as the bubble for consistency.
- Fully rounded corner, so it's not a "conversation bubble" but rather a "voiceover bubble".
- Light gray background: slightly visible but not too noisy.
- No border, let's leave the border for real messages.
- Full width: it allows to take out the notes of the conversation.
![Capture d’écran 2022-07-29 à 09 03 13](https://user-images.githubusercontent.com/80679690/181741724-e2dcb03c-8e29-4856-9469-a3e9363ca0b6.png)

I also improved the design of the TrackingValue in a separate commit.

alexkuhn reviewed (approved):
@robodoo r+ rebase-ff

robodoo commented:
Merge method set to rebase and fast-forward
